### ISSUE TITLE
[🐞] <Link> Does Not Work With Storybook 

### ISSUE THREAD

**Issue by thenhawke** [BuilderIO/qwik#5684]
### Which component is affected?

Starters / CLI

### Describe the bug

Using a \<Link\> in a story throws an error and breaks rendering in Storybook. If the \<Link\> components are removed, everything works as expected. 

Thrown error is as follows:

```
Uncaught Error: Code(13): Actual value for useContext() can not be found, make sure some ancestor component has set a value using useContextProvider()
    at createAndLogError (core.mjs?v=455dff6e:159:54)
    at logErrorAndStop (core.mjs?v=455dff6e:108:17)
    at qError (core.mjs?v=455dff6e:258:12)
    at useContext (core.mjs?v=455dff6e:3536:11)
    at useNavigate (index.qwik.mjs?v=455dff6e:357:25)
    at AsyncFunction.Link_component_8gdLBszqbaM [as resolved] (index.qwik.mjs:983:15)
    at AsyncFunction.invokeApply (core.mjs?v=455dff6e:4390:26)
    at AsyncFunction.invoke (core.mjs?v=455dff6e:4382:24)
    at core.mjs?v=455dff6e:8439:35
    at async Promise.all (:6006/index 0)
```

### Reproduction

https://github.com/thenhawke/qwik-link-component-storybook-example

### Steps to reproduce

Clone the repository
Run `npm install`
Run `npm run storybook`
Navigate to the "Example Link" "Primary" story, note the error in the console

### System Info

```shell
System:
  OS: macOS 14.2.1
  CPU: (6) x64 Intel(R) Core(TM) i5-8500B CPU @ 3.00GHz
  Memory: 829.64 MB / 32.00 GB
  Shell: 5.9 - /bin/zsh
Binaries:
  Node: 18.18.0 - ~/.nvm/versions/node/v18.18.0/bin/node
  npm: 9.8.1 - ~/.nvm/versions/node/v18.18.0/bin/npm
Browsers:
  Brave Browser: 120.1.61.104
  Chrome: 120.0.6099.199
  Safari: 17.2.1
npmPackages:
  @builder.io/qwik: ^1.3.2 => 1.3.2 
  @builder.io/qwik-city: ^1.3.2 => 1.3.2 
  undici: ^5.26.0 => 5.28.2 
  vite: ^5.0.6 => 5.0.11
```

### Additional Information

The example repository was created by running `npm create qwik@latest` followed by `npm run qwik add storybook`, then creating a story with a \<Link\> component in it and then running `npm run storybook`. Navigating to a component with the \<Link\> component in it will not render the component. 

I think this is the same issue as #5346 but I created an example repository.

**Comment by wmertens** [comment]
The reason is that the link component needs context that is automatically injected by the qwikcity component

https://github.com/BuilderIO/qwik/blob/d168a85dbc1681145373c5fe2b938e7ee9d33869/packages/qwik-city/runtime/src/link-component.tsx#L15C22-L15C22

Can you add that component to the storybook rendering?

**Comment by gioboa** [comment]
@thenhawke did you try with [QwikCityMockProvider](https://qwik.builder.io/docs/api/#qwikcitymockprovider )?

**Comment by thenhawke** [comment]
@wmertens @gioboa Yes, the context was missing. Adding a \<QwikCityMockProvider\> is exactly what was needed.

I had searched 'context' and 'useContext' in the docs before I made the issue but nothing about the mock provider showed up, I just somehow totally missed that part of the API. Thanks for the help!

**Comment by gioboa** [comment]
We can improve the StoryBook documentation page. Can I ask you to drop a PR for that pls?
